src/main/java/org/mockbukkit/mockbukkit/block/data/AgeableDataMock.java

```diff
@@ -37,4 +47,11 @@ public int getMaximumAge()
 		return this.getLimitationValue(BlockDataLimitation.Type.MAX_AGE);
 	}
 
+	@Override
+	@SuppressWarnings({"java:S2975", "java:S1182"})
+	public @NotNull AgeableDataMock clone()
+	{
+		return new AgeableDataMock(this);
```

Comment:
this looks super repetitive. And easy to forget when adding a new subclass.
Couldn't you do something in the parent like "this.getClass()::new(this)"?

Reply:
The benefit of this is that you don't have to implement it everywhere, and if the copy constructor is not implemented, it will fail for that class.

Reply:
Yeah, but then the user needs to implement that interface. There needs to be some kind of check somewhere. Because I'm sure sooner rather than later someone is going to forget it...
I tend to agree with you on reflection, but I didn't know how else to enforce it. Precondition somewhere? Or looping the files (if there is no global registry available) and checking in a test they do implement that copy constructor?

Reply:
I think the best approach is like you suggested, iterate each matearial that is a block (isBlock() method) and execute the clone operation

Reply:
Some things are in registries, so you could lean on that. And say that it needs to implement the interface. Compile-time protection. But others you cannot do that sadly enough. Or at least not that I can see.